src/Button/Button.tsx

```diff
@@ -40,33 +42,41 @@ export const Button = React.forwardRef<HTMLElement, ButtonProps>(
         $size={size}
         $variant={variant}
         $hasChildren={hasChildren}
-        disabled={disabled}
+        $isLoading={isLoading}
+        disabled={disabled || isLoading}
         href={href}
         onClick={onClick}
         {...rest}
         ref={ref}
       >
-        {startIcon && (
-          <Box display="flex" mr={hasChildren ? 1 : undefined}>
-            <Icon icon={startIcon} size={sizeIcon[size]} />
-          </Box>
-        )}
-        {hasChildren && (
-          <StyledTypography
-            $color={color}
-            $variant={variant}
-            disabled={disabled}
-            variant={TYPOGRAPHY_VARIANT_MAPPING[size]}
-            forwardedAs="span"
-          >
-            {children}
-          </StyledTypography>
-        )}
-        {endIcon && (
-          <Box display="flex" ml={hasChildren ? 1 : undefined}>
-            <Icon icon={endIcon} size={sizeIcon[size]} />
-          </Box>
+        {isLoading && (
+          <StyledSpinnerWrapper>
+            <Spinner size={SPINNER_VARIANT_MAPPING[size]} />
+          </StyledSpinnerWrapper>
         )}
+
+        <StyledContent isLoading={isLoading}>
+          {startIcon && (
+            <Box display="flex" mr={hasChildren ? 1 : undefined}>
+              <Icon icon={startIcon} size={sizeIcon[size]} />
+            </Box>
+          )}
+          {hasChildren && (
+            <StyledTypography
+              $variant={variant}
+              disabled={disabled}
+              variant={TYPOGRAPHY_VARIANT_MAPPING[size]}
+              forwardedAs="span"
+            >
```

Comment:
For this Typography, and looking at the Button spec from figma... are we missing the prop `fontWeight={700}` maybe?